columnflow/selection/cms/jets.py

```diff
@@ -53,61 +54,64 @@ def jet_veto_map(
 
     documentation: https://cms-jerc.web.cern.ch/Recommendations/#jet-veto-maps
     """
-
     jet = events.Jet
     muon = events.Muon[events.Muon.isPFcand]
 
     jet_mask = (
         (jet.pt > 15) &
         (jet.jetId >= 2) &
         (jet.chEmEF < 0.9) &
-        (ak.all(events.Jet.metric_table(muon) >= 0.2, axis=2))
+        ak.all(events.Jet.metric_table(muon) >= 0.2, axis=2)
     )
 
     # apply loose Jet puId in Run 2 to jets with pt below 50 GeV
-    if self.config_inst.campaign.x.year <= 2018:
+    if self.config_inst.campaign.x.run == 2:
         jet_pu_mask = (events.Jet.puId >= 4) | (events.Jet.pt > 50)
         jet_mask = jet_mask & jet_pu_mask
 
     # for some reason, math.pi is not included in the ranges, so we need to subtract a small number
     pi = math.pi - 1e-10
 
     # values outside [-pi, pi] are not included, so we need to wrap the phi values
-    jet_phi = ak.where(np.abs(events.Jet.phi) > pi, events.Jet.phi - 2 * pi * np.sign(events.Jet.phi), events.Jet.phi)
-
+    jet_phi = ak.where(
+        np.abs(events.Jet.phi) > pi,
+        events.Jet.phi - 2 * pi * np.sign(events.Jet.phi),
+        events.Jet.phi,
+    )
     variable_map = {
         "type": "jetvetomap",
         "eta": jet.eta,
         "phi": jet_phi,
     }
-
     inputs = [variable_map[inp.name] for inp in self.veto_map.inputs]
 
-    # apply the veto map
-    veto_map_result = ak.where(
-        jet_mask,
-        self.veto_map(*inputs),
-        -1,
-    )
-
-    # get the Jet veto mask (events containing such a jet should be vetoed)
-    veto_map_jet_mask = (veto_map_result > 0)
-
-    if self.config_inst.campaign.x("postfix", "").lower() == "bpix":
-        # in postBPix, we need to run the veto map with type=jetvetomap_bpix and subtract this from
-        # the result of the nominal jet veto map
-        raise NotImplementedError("Jet Veto Map for 2023 postBPix not implemented yet")
+    # apply the veto map to selected jets
+    # (a map value of 0 means the jet is to be vetoed)
+    veto_mask = jet_mask & (self.veto_map(*inputs) == 0)
```

Comment:
No, this is wrong. An event is to be vetoed when a jet with veto map value > 0 survives the jet preselection.
```suggestion
    # (a map value of >0 means the jet is to be vetoed)
    bad_jet = jet_mask & (self.veto_map(*inputs) > 0)
    good_jet = jet_mask | (self.veto_map(*inputs) == 0)
```

Afterwards, we need to define the selection step accordingly
```
# check that no bad jets are present or that all jets are good jets.
jet_veto_map_step = ak.sum(bad_jet, axis=1) == 0
jet_veto_map_step = ak.sum(good_jet, axis=1) == ak.num(good_jet)
```

Reply:
I fixed it, please have a look :+1: